kitsune/search/v2/views.py

```diff
@@ -0,0 +1,114 @@
+import json
+
+from django.conf import settings
+from django.http import HttpResponse
+from django.utils.translation import ugettext as _, pgettext
+
+from kitsune import search as constants
+from kitsune.search.forms import SimpleSearchForm
+from kitsune.sumo.api_utils import JSONRenderer
+from kitsune.sumo.utils import build_paged_url
+from kitsune.products.models import Product
+from kitsune.search.views import _fallback_results
+from kitsune.search.utils import locale_or_default
+from kitsune.search.v2.search import Search, KBConfig, AAQConfig
+
+
+def simple_search(request):
+    search_form = SimpleSearchForm(request.GET, auto_id=False)
+
+    if not search_form.is_valid():
+        return HttpResponse(
+            json.dumps({"error": _("Invalid search data.")}),
+            content_type="application/json",
+            status=400,
+        )
+
+    cleaned = search_form.cleaned_data
+
+    # get language
+    language = locale_or_default(cleaned["language"] or request.LANGUAGE_CODE)
+    lang_name = settings.LANGUAGES_DICT.get(language.lower()) or ""
+
+    # get product
+    product = Product.objects.filter(slug__in=cleaned["product"]).first()
+    if product:
+        product_titles = [pgettext("DB: products.Product.title", product.title)]
+    else:
+        product_titles = [_("All Products")]
+
+    # get page
+    try:
+        page = int(request.GET.get("page", 1))
+    except ValueError:
+        page = 1
+
+    # create search object
+    search = Search(cleaned["q"], language, product=product, page=page)
+
+    # apply aaq/kb configs
+    if cleaned["w"] & constants.WHERE_WIKI:
+        search.add_config(KBConfig)
+    if cleaned["w"] & constants.WHERE_SUPPORT:
+        search.add_config(AAQConfig)
+
+    # execute search
+    search.execute()
+    total = search.total
+    results = search.results
+
+    # generate fallback results if necessary
+    fallback_results = None
+    if total == 0:
+        fallback_results = _fallback_results(language, cleaned["product"])
+
+    # create results dictionary for instant search
+    data = {
+        "num_results": total,
+        "total": total,
+        "results": results,
+        "fallback_results": fallback_results,
+        "product_titles": product_titles,
+        "q": cleaned["q"],
+        "w": cleaned["w"],
+        "lang_name": lang_name,
+        "products": [
+            {"slug": p.slug, "title": p.title} for p in Product.objects.filter(visible=True)
+        ],
+        "pagination": _make_pagination(request, page, total),
+    }
+    if product:
+        data["product"] = product.slug
+    if not results:
+        data["message"] = _("No pages matched the search criteria")
+
+    json_data = JSONRenderer().render(data)
+    return HttpResponse(json_data, content_type="application/json")
+
+
+def _make_pagination(request, page, total):
```

Comment:
why do we need this?

Reply:
It's needed to render the pagination section at the bottom of the instant search